### PR TITLE
Add configuration for StyleCop rules

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -269,8 +269,8 @@ dotnet_diagnostic.SA1028.severity = warning
 
 # Ordering Rules (SA1200-) https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/1.1.118/documentation/OrderingRules.md
 
-dotnet_diagnostic.SA1200.severity = none # needs configuration to be "outsideNamespace"
-dotnet_diagnostic.SA1201.severity = none # needs configuration for our coding style order
+dotnet_diagnostic.SA1200.severity = warning
+dotnet_diagnostic.SA1201.severity = warning # needs monitoring, might need tuning inside the stylecop.json configuration
```
Comment (pavel-mikula-sonarsource) [inline .editorconfig:273]: This one complains about our constructors => too noisy

File: analyzers/stylecop.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
+  "settings": {
+    "orderingRules": {
+      "systemUsingDirectivesFirst": true,
```
Comment (pavel-mikula-sonarsource) [inline analyzers/stylecop.json:5]: `true` is default => we should not need it
```suggestion
```

File: analyzers/stylecop.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      "usingDirectivesPlacement": "outsideNamespace",
+      "elementOrder": [
+        "constant",
+        "static",
```
Comment (pavel-mikula-sonarsource) [inline analyzers/stylecop.json:9]: We should ignore the `static` configuration, as it's noisy for private static methods in our code base.
```suggestion
```

File: analyzers/stylecop.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        "accessibility",
+        "readonly",
+        "kind"
```
Comment (pavel-mikula-sonarsource) [inline analyzers/stylecop.json:12]: Open question: Does it make sense to have `kind` configured when we can't use SA1201 (yet)?

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/stylecop.json (hunk 3 above)
Comment (andrei-epure-sonarsource) [reply analyzers/stylecop.json:9]: yes, and I will also remove accessibility, because we can have `private static` at the top

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**